### PR TITLE
Update comments in column_family.h

### PR DESCRIPTION
**Pull request by siying**
Document relationships of data structures declared in column_family.h

### CODE REVIEW

File: db/column_family.h [hunk 1]
```diff
@@ -45,7 +45,113 @@ class InstrumentedMutexLock;
...
 extern const double kIncSlowdownRatio;
-
+// This file contains a list of data structures for managing column family
+// level metadata. 
+//
+// The basic relationships among classes declared here are illustrated as
+// following:
+//
+//       +----------------------+    +----------------------+   +--------+
+//   +---+ ColumnFamilyHandle 1 | +--+ ColumnFamilyHandle 1 |   | DBImpl |
```
Comment (riversand963) [inline db/column_family.h]: The column family handle in the middle should be column family handle 2.

File: db/column_family.h [hunk 2]
```diff
@@ -45,7 +45,113 @@ class InstrumentedMutexLock;
...
+//              |      1       +------------------------------>+            |
+//              |              +-+  |        mem               | MemTable b |
+//              +--------------+ |  |                          |            |
+//                               |  |    +--------------+      +-----^------+
+//                               |  |imm | MemtableList |            |
+//                               |  +--->+   Version s  +------------+
+//                               |       +--------------+
+//                               |       +--------------+
+//                               |       | MemtableList |
+//                               +------>+   Version t  +-------->  empty
```
Comment (riversand963) [inline db/column_family.h]: s/empty/null/

File: db/column_family.h [hunk 3]
```diff
@@ -45,7 +45,113 @@ class InstrumentedMutexLock;
...
+// unreferenced to 0 and destoryed.
+//
+// The following graph illustrates a possible referencing relationships:
+//
+// Column       +--------------+      current       +-----------+
+// Family +---->+              +------------------->+           |
+//  Data        | SuperVersion +----------+         | Version A |
+//              |      3       |   imm    |         |           |
+// Iter2 +----->+              |  +-------v------+  +-----------+
+//              +-----+--------+  | MemtableList +----------------> Empty
```
Comment (riversand963) [inline db/column_family.h]: s//Empty/null?

File: db/column_family.h [hunk 4]
```diff
@@ -45,7 +45,113 @@ class InstrumentedMutexLock;
...
+//                               |  |    +--------------+      +-----^------+
+//                               |  |imm | MemtableList |            |
+//                               |  +--->+   Version s  +------------+
+//                               |       +--------------+
+//                               |       +--------------+
+//                               |       | MemtableList |
+//                               +------>+   Version t  +-------->  empty
+//                                 imm   +--------------+
+//
+// In this example, even if the current LSM-tree is consist of Version A and
```
Comment (riversand963) [inline db/column_family.h]: consists of

File: db/column_family.h [hunk 5]
```diff
@@ -45,7 +45,113 @@ class InstrumentedMutexLock;
...
+// memtable a, which is also referenced by SuperVersion, two older SuperVersion
+// SuperVersion2 and Superversion1 still exists, and are referenced by a
```
Comment (riversand963) [inline db/column_family.h]: exist

File: db/memtable_list.h [hunk 6]
```diff
@@ -27,6 +27,7 @@
 
 namespace rocksdb {
 
+
```
Comment (riversand963) [inline db/memtable_list.h]: Is this extra space intentional?

File: db/column_family.h (hunk 3 above)
Comment (siying) [reply db/column_family.h]: I debated myself about it. My reasoning is that, the _list_ is _empty_, rather than pointing to a memtable that is _null_.

Comment (riversand963) [reply db/column_family.h]: I am actually fine with either. If we go with 'empty', then either they should both be 'Empty' or 'empty'.

File: db/column_family.h [hunk 7]
```diff
@@ -45,7 +45,113 @@ class InstrumentedMutexLock;
...
+//                    |           |   Version r  |  +-----------+
+//                    |           +--------------+  |           |
+//                    +------------------+   current| Version B |
+//              +--------------+         |   +----->+           |
+//              |              |         |   |      +-----+-----+
+// Compaction +>+ SuperVersion +-------------+            ^
+//    Job       |      2       +------+  |                |current
+//              |              +----+ |  |     mem        |    +------------+
+//              +--------------+    | |  +--------------------->            |
+//                                  | +------------------------> MemTable a |
```
Comment (sagar0) [inline db/column_family.h]: This line should also be marked mem.

**Pull request closed without merging**